### DISCUSSION TITLE
How to Scale X Values to Fit Entirely in Cartesian Chart

### DISCUSSION THREAD

**Discussion by mak1nt0sh** [patrykandpatrick/vico#733] [Questions] [answered]
I'm currently using the Vico 2 library to create a Cartesian chart, but I'm encountering an issue where the x-values are stretched and not fitting within the chart properly. I've read that setting `rememberVicoScrollState(scrollEnabled = false)` should intuitively scale the chart to fit all values, but it's not working as expected.

Here's a snippet of my code:

```
lineSeries {
    series(x = historyXEpochDays.keys, y = graphData.historyGraphData.values)
    series(x = forecastXEpochDays.keys, y = graphData.forecastGraphData.values)
}

CartesianChartHost(
    modifier = modifier,
    chart = rememberCartesianChart(
        rememberLineCartesianLayer(
            lines = listOf(
                rememberLineSpec(DynamicShader.color(riskColor)),
                rememberLineSpec(DynamicShader.color(riskColor.copy(alpha = 0.6f)))
            )
        ),
        startAxis = rememberStartAxis(),
        bottomAxis = rememberBottomAxis()
    ),
    modelProducer = budgetGraphModeProducer,
    scrollState = rememberVicoScrollState(scrollEnabled = false),
    zoomState = rememberVicoZoomState(zoomEnabled = false)
)
```

**Comment by patrickmichalik** [comment, marked as answer]
Hello! The automatic behavior in question is tied to the default value of the `zoomState` parameter. Since you’re passing in a custom `VicoZoomState` instance, it’s overridden. Two solutions are available.

The simplest option is to remove the `zoomState` argument. When scroll is disabled, zoom is too. (Notably, however, you can now create a chart that makes all of its content initially visible and can then be zoomed and scrolled. This wasn’t the case in Vico 1.)

Alternatively, you can use the following:

```kt
rememberVicoZoomState(zoomEnabled = false, initialZoom = Zoom.Content)
```
